src/LightSaml/SpBundle/Infrastructure/RestApi.php

```
@@ -0,0 +1,42 @@
+<?php
+/*
+ * This file is part of the LightSAML SP-Bundle package.
+ *
+ * (c) Milos Tomic <tmilos@lightsaml.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+namespace LightSaml\SpBundle\Infrastructure;
+
+use GuzzleHttp\Psr7\Request;
+use Http\Client\Exception;
+use Http\Client\HttpClient;
+
+abstract class RestApi
+{
+    /** @var HttpClient */
+    private $httpClient;
+
+    /** @var string */
+    private $restApiUrl;
+
+    public function __construct(HttpClient $httpClient, $restApiUrl)
+    {
+        $this->httpClient = $httpClient;
+        $this->restApiUrl = $restApiUrl;
+    }
+
+    protected function sendRequest($method, $resourcePath)
+    {
+        try {
+            return $this->httpClient->sendRequest(new Request(
+                $method,
+                sprintf('%s/%s', $this->restApiUrl, $resourcePath),
+                ['Content-Type' => 'application/json']
+            ));
+        } catch (Exception $e) {
```

I would suggest to import it as HttpClientException - that way it's easier to understand what exception you are trying to catch.

---

This method exceptions are as follows
```
     * @throws \Http\Client\Exception If an error happens during processing the request.
     * @throws \Exception             If processing the request is impossible (eg. bad configuration).
```
So afaik @chriskicker wanted to catch all exceptions here.

---

`use Http\Client\Exception;` - this is what he's catching
And I suggest to
`use Http\Client\Exception as HttpClientException;`
then
`catch (HttpClientException $e) {`
Seems more readable to me but I'm not a stubborn man - if you prefer to leave it as is I won't mind.

---

Indeed I didn't noticed that. Should be refined in a sec.